lib/Core/injectTerms.ts

```diff
@@ -0,0 +1,117 @@
+import { Term } from "../ReactViewModels/defaultTerms";
+
+const findFirstTerm = (
+  text: string,
+  terms: Map<string, Term>,
+  fromIndex: number
+) => {
+  let termIndex = Infinity;
+  let termLength = 0;
+  let termToReplace: string | undefined;
+  let ignore = false;
+
+  terms.forEach((_, term) => {
+    const foundIndex = text
+      .toLowerCase()
+      .indexOf(term.toLowerCase(), fromIndex);
+    if (
+      foundIndex !== -1 &&
+      (foundIndex < termIndex ||
+        (foundIndex <= termIndex && term.length > termLength))
+    ) {
+      const nextOpeningBrace = text.toLowerCase().indexOf("[", foundIndex);
+      const nextClosingBrace = text.toLowerCase().indexOf("]", foundIndex);
+      const hasNextClosingBrace = nextClosingBrace !== -1;
+      const nextBracesAreClosedLink =
+        nextOpeningBrace !== -1 &&
+        hasNextClosingBrace &&
+        nextOpeningBrace < nextClosingBrace;
+      const inLinkLabel = hasNextClosingBrace && !nextBracesAreClosedLink;
+
+      const nextOpeningParen = text.toLowerCase().indexOf("(", foundIndex);
+      const nextClosingParen = text.toLowerCase().indexOf(")", foundIndex);
+      const hasNextClosingParen = nextClosingParen !== -1;
+      const nextParensAreClosedLink =
+        nextOpeningParen !== -1 &&
+        hasNextClosingParen &&
+        nextOpeningParen < nextClosingParen;
+      const inLinkHref = hasNextClosingParen && !nextParensAreClosedLink;
```

Comment:
Yeah you're right - it's a lazy check, should ideally check it's not immediately next to a brace/paren etc.
Do you think it would be OK to do this in a separate PR?

Reply:
Yup, given that the deadline is near I think it's fine to do it later in a separate PR, since we probably won't come across this edge case that often